server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1436,11 +1438,13 @@ public void processPendingDeletes(Index index, IndexSettings indexSettings, Time
                     }
                     if (remove.isEmpty() == false) {
                         logger.warn("{} still pending deletes present for shards {} - retrying", index, remove.toString());
-                        Thread.sleep(sleepTime);
+                        if (stopLatch.await(sleepTime, TimeUnit.MILLISECONDS)) {
+                            break;
```

Comment:
Maybe log here too?

Reply:
Added logging in 71146a264a9595289cd1fce66c49ae9f829882f2